### PR TITLE
Catch errors to avoid production env deployment failure

### PR DESCRIPTION
**Pull request by aaron-collier**
## Why was this change made?

This adds the rescue for the general rspec to the no-prompts setup that was recently added in order to avoid deployment errors via capistrano.

## Was the documentation (README, DevOpsDocs, wiki, openapi.json) updated?

N/A

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -25,9 +25,14 @@ rescue LoadError
   puts 'gem rspec not available presumably b/c this is a production environment'
 end
 
-desc "Run all specs that do not prompt user for input'"
-RSpec::Core::RakeTask.new(:spec_ci) do |t|
-  t.rspec_opts = '--tag ~prompts'
+begin
```
Comment (mjgiarlo) [inline Rakefile]: Should this be merged into the code on lines 22-26?

Comment (mjgiarlo) [reply Rakefile]: OK, I think lines 30-33 should move to line 24 above.

Comment (aaron-collier) [reply Rakefile]: @mjgiarlo fixed

File: Rakefile [hunk 2]
```diff
@@ -21,14 +21,13 @@ end
 
 begin
   require 'rspec/core/rake_task'
+  desc "Run all specs that do not prompt user for input'"
+  RSpec::Core::RakeTask.new(:spec_ci) do |t|
+    t.rspec_opts = '--tag ~prompts' unless Rails.env.production?
+  end
```
Comment (mjgiarlo) [inline Rakefile]: @aaron-collier While not strictly required for this PR, I wonder if you'd be open to moving these lines here as well: 

```ruby
  desc 'Run Continuous Integration Suite (linter and tests)'
  task ci: %i[rubocop db:migrate spec_ci]
 
  # and replace it with our own
  task default: :ci
```

Else we've got undefined tasks in `production`. Not a huge deal. (`task(:default).clear` can/should stay where it is

Comment (aaron-collier) [reply Rakefile]: Done.

Comment (ndushay) [reply Rakefile]: meh - it kinda weirds me out to have the default task not always available ... but not enough to make a PR to put it outside the block.

That's why I put all that stuff outside the block in the first place.

No perfect solution.  I defer.

**Pull request merged**